nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/WebApplicationTargetsConditionPatcher.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Language.Xml;
+
+namespace NuGetUpdater.Core.Updater
+{
+    internal class WebApplicationTargetsConditionPatcher : IDisposable
+    {
+        private string? _capturedCondition;
+        private readonly XmlFilePreAndPostProcessor _processor;
+
+        public WebApplicationTargetsConditionPatcher(string projectFilePath)
```

Comment:
I like this shape, I think it makes it relatively easy to create a couple of similar Patchers and group them together into an aggregator.